cogs/tags/tags.py

```diff
@@ -136,6 +157,9 @@ def __init__(self, bot):
         self.settings = Config("settings.json", cogname="tags")
         self.configV3 = ConfigV3.get_conf(self, identifier=5842647, force_registration=True)
         self.configV3.register_guild(**BASE)  # Register default (empty) settings.
+        for guildID, _ in self.configV3.all_guilds():
+            with self.configV3.guild(guildID).tiers() as tiers:
+                self.allowed_roles[guildID] = tiers.keys()
         self.lock = Lock()
```

Comment:
Few things here:
1. `self.configV3.all_guilds()` is a coroutine, so method here must be labeled async.
2. Because of 1 and constructor can't be a coroutine, we need to have this separate.
3. `self.configV3.guild(...)` accepts a discord.Guild object, so we must convert ID to this type.

```suggestion
        self.lock = Lock()
        self.bot.loop.create_task(self.syncAllowedRoles())

    async def syncAllowedRoles(self):
        guildIds = await self.configV3.all_guilds()
        for guildId in guildIds.keys():
            guild = discord.utils.get(self.bot.guilds, id=guildId)
            async with self.configV3.guild(guild).tiers() as tiers:
                self.allowed_roles[guildId] = tiers.keys()
```